src/ln/peer_handler.rs

```diff
@@ -121,6 +121,16 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 		}
 	}
 
+	/// Get the list of public keys of peers with initial handshake completed
+	pub fn get_peer_node_ids(&self) -> Option<Vec<PublicKey>> {
+		let peers = self.peers.lock().unwrap();
+		if peers.node_id_to_descriptor.is_empty() {
```

Comment:
If you want to differentiate "no peers connected" with "no peers have finished the noise handshake", then that should at least be documented, though I'm skeptical there's much value in that distinction (in part because the client always knows whether it has connections open).

Reply:
Updated to take pubkeys that have been learned.